hts.c

```diff
@@ -2254,14 +2254,17 @@ int hts_itr_next(BGZF *fp, hts_itr_t *iter, void *r, void *data)
 /**********************
  *** Retrieve index ***
  **********************/
-
-static char *test_and_fetch(const char *fn)
+// Returns -1 if index couldn't be opened.
+//         -2 on other errors
+static int test_and_fetch(const char *fn, const char **local_fn)
 {
+    hFILE *fp_remote;
```

Comment:
The naming of the variables is a bit inconsistent. Could we have either `fp_remote` and `fn_local` or `remote_fp` and `local_fn`?

Reply:
Agreed.  I've pushed a new commit to make them more consistent.